### PR TITLE
Update all routes (header, footer, dashboard gallery) to dashboard dynamic routes (from API). Removed topics routes.

### PR DESCRIPTION
**Pull request by dancer2090**
## Overview
This PR is the second part to move topics to dashboards:
1) This PR updates the routes for dashboards gallery. Topics routes removed from dashboards page and menus.
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/170311468)
2)  This PR updates the links in header and footer. Topic links were changed to feature dashboards links.
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/170311607)
3) This PR updates API requests for featured dashboard and dashboard gallery. Added is-featured and is-highlighted filter options to requests. (updated requests and redux state).
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/170311599)
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/169732990)
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/170397448)
4) This PR updates visual part of dashboard gallery.
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/169732990)
5) This PR removes and renames old topics files.
[Pivotal task](https://www.pivotaltracker.com/n/projects/1374154/stories/170311546)
## Testing instructions
Use staging API
1) Footer and Header links:
 - Go to header menu, hover dashboards, click on dashboard.
 - Go to footer menu click on dashboard.
You should go to /dashboards/:slug page
2) Gallery links:
 - Go to /dashboads page. Click on gallery item. You should go to /dashboards/:slug page
3) Featured links
 - Go to /dashboads page. Click on item in featured dashboads list. You should go to /dashboards/:slug page
The featured option has already added to a request (for links and featured dashboards). But is not deployed on staging on BE side. We shoul wait utill this option will be activated.
4) Is-highlighted option. This option was added on staging by previous PR. And added to admin dashboard (new/edit) pages too. So, to add items to dashboard gallery go to /admin/dashboards select or create dashboard and set `is-highlighted` and `published` flags
---

## Checklist before submitting
- [x] Title: Don't forget to make it clear for everyone, even for people that don't know about the feature/bug.
- [x] Meaningful commits and code rebased on `develop`.

### CODE REVIEW

File: components/dashboards/thumbnail-list/component.js [hunk 1]
```diff
@@ -1,32 +1,69 @@
...
+                })}
+              >
+                <button
+                  key={dashboard.slug}
+                  tabIndex="0"
+                  className={buttonClass}
+                  style={{ backgroundImage: `url(${dashboard.photo && dashboard.photo.cover})` }}
+                  onClick={() => onSelect(dashboard)}
+                >
+                  <div className="content" htmlFor={`topic-${dashboard.slug}`}>
```
Comment (pablopareja) [inline components/dashboards/thumbnail-list/component.js]: The `htmlFor` attribute content should be updated _(we should remove the "topic" prefix and replace it)_

File: components/dashboards/thumbnail-list/component.js [hunk 2]
```diff
@@ -1,32 +1,69 @@
...
-            <Link
-              to="dashboards_detail_custom"
-              params={{ slug: _dashboard.slug }}
-            >
-              <a><span>{_dashboard.name}</span></a>
-            </Link>
-          </li>
-        ))}
-      </ul>
+      <div className="c-dashboards-gallery-list">
```
Comment (pablopareja) [inline components/dashboards/thumbnail-list/component.js]: Why did you change the name of the CSS class used to a different value that doesn't correspond to the name of the component?
Besides, I've seen that the former CSS class `c-dashboard-thumbnail-list` is no longer used anywhere in the code. 
Why didn't you replace the old CSS code with the one that you put in the new CSS file called `dashboards_gallery.scss`?

File: css/components/dashboards/dashboards-gallery.scss [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  flex-wrap: wrap;
+  margin: 0;
+  padding: 0;
+  list-style: none;
+
+  .thumbnail-list-item {
+    position: relative;
+    height: 200px;
+    border-radius: 4px;
+    background-color: $color-dark-grey;
```
Comment (pablopareja) [inline css/components/dashboards/dashboards-gallery.scss]: Please use `$blue-grey` here instead _(it holds the same value ,but we should take as a reference the CSS vars defined in the file `_settings.scss` and avoid using the ones defined in the file `_deprecated.scss`)_

File: css/components/dashboards/dashboards-gallery.scss [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    padding: 0;
+    transition: all $animation-time-2 $ease-in-out-sine;
+
+    &:nth-of-type(6n) {
+      margin-right: 0;
+    }
+
+    // The input is just here for an accessibility matter
+    // we don't want it to be visually present
+    input[type="radio"] {
```
Comment (pablopareja) [inline css/components/dashboards/dashboards-gallery.scss]: I don't see any `input` included as part of the dashboard gallery, thus I don't understand this CSS rule definition here.

File: css/components/dashboards/dashboards-gallery.scss [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        
+        .dashboard-title {
+          color: #393F44;
+          font-family: Lato;
+          font-size: 16px;
+          font-weight: bold;
+          line-height: 20px;
+          text-align: left;
+          padding: 15px 15px 10px 15px;
```
Comment (pablopareja) [inline css/components/dashboards/dashboards-gallery.scss]: Please use the `$space...` CSS variables defined in `_settings.scss` for this and any other similar cases 😉 .

File: css/components/dashboards/dashboards-gallery.scss [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      
+      .text {
+        width: 100%;
+        min-height: 75px;
+        background-color: $white;
+
+        
+        .dashboard-title {
+          color: #393F44;
+          font-family: Lato;
```
Comment (pablopareja) [inline css/components/dashboards/dashboards-gallery.scss]: This isn't necessary, we already have it defined through the `_settings.scss` CSS variable `$body-font-family`

File: css/index.scss [hunk 7]
```diff
@@ -244,6 +244,7 @@
 @import './components/dashboards/list';
 @import './components/dashboards/template-selector';
 @import './components/dashboards/dashboards-subheader-block';
+@import './components/dashboards/dashboards-gallery';
```
Comment (pablopareja) [inline css/index.scss]: This is probably not necessary if we reuse the `dashboard-thumbnail-list` styles file.

File: components/dashboards/thumbnail-list/component.js (hunk 2 above)
Comment (dancer2090) [reply components/dashboards/thumbnail-list/component.js]: dashboards_gallery renamed to c-dashboards-thumbnail-list. fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/906f1bce4b882a0b0bfb84a641261c884020b757

File: css/components/dashboards/dashboards-gallery.scss (hunk 3 above)
Comment (dancer2090) [reply css/components/dashboards/dashboards-gallery.scss]: Fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/68daf5ece578aa9e34947bc6518d91f784da794a

File: css/components/dashboards/dashboards-gallery.scss (hunk 4 above)
Comment (dancer2090) [reply css/components/dashboards/dashboards-gallery.scss]: Fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/68daf5ece578aa9e34947bc6518d91f784da794a

File: css/components/dashboards/dashboards-gallery.scss (hunk 5 above)
Comment (dancer2090) [reply css/components/dashboards/dashboards-gallery.scss]: Fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/68daf5ece578aa9e34947bc6518d91f784da794a

File: css/components/dashboards/dashboards-gallery.scss (hunk 6 above)
Comment (dancer2090) [reply css/components/dashboards/dashboards-gallery.scss]: Fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/6e9c6fca9edce03d58fe1783a0297f657b0c0b69

File: components/dashboards/thumbnail-list/component.js (hunk 1 above)
Comment (dancer2090) [reply components/dashboards/thumbnail-list/component.js]: Fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/4078f99678fab223b1f93fce6c7390e3a117ae5c

File: css/index.scss (hunk 7 above)
Comment (dancer2090) [reply css/index.scss]: Fixed here: https://github.com/resource-watch/resource-watch/pull/1185/commits/906f1bce4b882a0b0bfb84a641261c884020b757

**Comment by dancer2090** [comment]
Lates commits remove and rename topics files to dashboards (for the new dashboard page)
